### PR TITLE
Port Mconf Timer into BBB 2.7

### PR DESCRIPTION
**Pull request by Arthurk12**
### What does this PR do?

This PR ports our Mconf Timer feature into BBB 2.7:

The timer feature is a useful tool that enables presenters or moderators to start a synchronized time counting across multiple clients.

The timer feature has 2 modes: Stopwatch and Timer itself.
- **Stopwatch** is the default one and it counts foward without a limit time.
- In **Timer** mode, set an initial time and it counts down to 0. If timer.alarm is true, an alarm will sound when the time ends.

There are 3 pre-selected songs that can be played along with the timer in Timer mode. They are configured in `settings.yml` file as well as their volume.
Also, if `timer.tabIndicator` is set to true in the `settings.yml` file the current time is displayed in the tab's title.

New flags added to the `settings.yml`:
```
  timer:
    enabled: true
    alarm: true
    music:
      enabled: true
      volume: 0.4
      track1: "RelaxingMusic"
      track2: "CalmMusic"
      track3: "aristocratDrums"
    interval:
      clock: 100
      offset: 60000
    time: 5
    tabIndicator: true
```

timer

### CalmMusic

Title: With You In My Arms
Artists: Edward Grenga (ASCAP), Charles Lawry (ASCAP), Douglas Stevens (ASCAP), Michael McMahon (ASCAP)
Album: Snappy & Smooth Jazz Music
https://www.sound-ideas.com/Product/291/Snappy-Smooth-Jazz-Music-Royalty-Free-Music

### RelaxingMusic

Title: Keeping Watch
Artists: Edward Grenga (ASCAP), Charles Lawry (ASCAP), Douglas Stevens (ASCAP), Michael McMahon (ASCAP)
Album: New Age Landscapes

https://www.sound-ideas.com/Product/297/New-Age-Landscapes-Royalty-Free-Music

### Happy

Title: Aristrocrat Drums
https://samplefocus.com/samples/aristocrat-drums-type-full-loop

### CODE REVIEW

**Comment by antobinary** [comment]
Wow! Well done! :clap:

File: bigbluebutton-html5/imports/ui/components/user-list/user-list-content/component.jsx [hunk 1]
```diff
@@ -35,6 +38,9 @@ class UserContent extends PureComponent {
         {isChatEnabled() ? <UserMessagesContainer /> : null}
         {currentUser.role === ROLE_MODERATOR ? <UserCaptionsContainer /> : null}
         <UserNotesContainer />
+        { isTimerActive && <TimerContainer
```
Comment (GuiLeme) [inline bigbluebutton-html5/imports/ui/components/user-list/user-list-content/component.jsx:41]: There are some ES-lint warnings in this file

File: bigbluebutton-html5/public/locales/en.json [hunk 2]
```diff
@@ -54,6 +54,26 @@
     "app.emojiPicker.skintones.4": "Medium Skin Tone",
     "app.emojiPicker.skintones.5": "Medium-Dark Skin Tone",
     "app.emojiPicker.skintones.6": "Dark Skin Tone",
+    "app.timer.title": "Time",
+    "app.timer.label": "Time",
+    "app.timer.stopwatch.title": "Stopwatch",
+    "app.timer.stopwatch.label": "Stopwatch",
+    "app.timer.timer.title": "Timer",
+    "app.timer.timer.label": "Timer",
```
Comment (GuiLeme) [inline bigbluebutton-html5/public/locales/en.json:62]:
```suggestion
    "app.timer.title": "Time",
    "app.timer.stopwatch.title": "Stopwatch",
    "app.timer.timer.title": "Timer",
```
Those are not used anymore

**Review by GuiLeme — CHANGES_REQUESTED** [review]
It is working just fine. I just made some suggestions regarding good practices. Overall great job!!

File: bigbluebutton-html5/imports/api/timer/index.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Meteor } from 'meteor/meteor';
+
+const Timer = new Mongo.Collection('timer');
+
+if (Meteor.isServer) {
+  Timer._ensureIndex({ meetingId: 1 });
```
[the remark this answers is not in the document]
Comment (antobinary) [reply bigbluebutton-html5/imports/api/timer/index.js:6]: Careful with the backtick ` in the suggested change. Otherwise - fair suggestion, thanks!

**Comment by antobinary** [comment]
FYI @GuiLeme is working on applying the suggested changes.

**Pull request merged**